replica/database.cc

```diff
@@ -1838,19 +1838,19 @@ future<> database::apply_in_memory(const mutation& m, column_family& cf, db::rp_
     return cf.apply(m, std::move(h), timeout);
```

Comment:
Under what conditions will we ever apply reversed mutations? Such writes will never come via CQL. Is this to support read repair during reverse reads? I think it is possibly simpler to just reverse the mutations we write to replicas as the last step, to avoid these very rarely used code-path changes to the write-path.

Reply:
Exactly for the read-repair during reverse reads. Reversing mutations just before they are sent, could do the work as well. Need to check it. In fact, we already reverse them when do have a mix-node cluster so it should work. I can reverse them always even for new nodes clusters.

Is this solution preferable over what is done now?

Reply:
Yes, I think sparing the write path from reverse-awareness is a worthy enough goal to always reverse just before sending the writes.

Reply:
Note that just before applying, the mutation has to be reversed anyway, because you cannot apply a reversed mutation directly.
So it is better to reverse on the coordinator and spare the replicas from this.

Reply:
Good catch. Yet another mixed-cluster testing nightmare. I agree sending only unreversed mutations in the write path is much better.

Perhaps we should unreverse before reconciliation to avoid yet another special-ness. I don't see anything special in reconciliation which wouldn't work with reversed mutations, but it's easy to overlook things.

Reply:
I think reconciliation works transparently with reverse reads, the only complication is unreversing mutations to be sent to replicas.